muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/discussion.js

```diff
@@ -217,63 +217,56 @@ $(document).ready(function() {
       element = element.parents(".di-messages-paging");
       var pageElement = $(".di-messages-pages");
       var _this = this;  
-      _this._addLoading(pageElement);
+      this._addLoading(pageElement);
       
       $(element).remove();
-      var msgsCount = 0;
-      var msgs = $(DiscImpl.msgContainer).find('.di-message');
+
+      var loadedMsgs = $(DiscImpl.msgContainer).find('.di-message');
+      var msgsCount = loadedMsgs.length;
       
-      for(var m = 0; m < msgs.length; m++){
-        msgsCount ++;
-       }
-            
-      var fRes = msgsCount;
       
       if (areaId == undefined){
-        mApi().forum.latest.read({'firstResult' : fRes}).on('$', function(msgs, msgsCallback) {
+        mApi().forum.latest.read({'firstResult' : msgsCount}).on('$', $.proxy(function(msgs, msgsCallback) {
           mApi().forum.areas.read(msgs.forumAreaId).callback(function(err, area) {
             msgs.areaName = area.name;
           });
-          
           var d = new Date(msgs.created);
           msgs.prettyDate = formatDate(d) + ' ' + formatTime(d);
           msgsCallback();
-        }).callback(function(err, threads) {
+        }, this)).callback($.proxy(function(err, threads) {
   
           if (err) {
             $('.notification-queue').notificationQueue('notification', 'error', getLocaleText('plugin.discussion.errormessage.nothreads', err));
           } else {
-
-            renderDustTemplate('/discussion/discussion_page.dust', threads, function(text) {
-              _this._clearLoading(); 
-             pageElement.append($.parseHTML(text));
+            threads.msgLoadCount = threads.length;
+	          this._clearLoading(); 
+        	  renderDustTemplate('/discussion/discussion_page.dust', threads, function(text) {
+                pageElement.append($.parseHTML(text));
             });
-          }
-        });     
+      }
+        }, this));     
       
       }else{
         mApi().forum.areas.threads.read(areaId, {'firstResult' : fRes}).on('$', function(thread, threadCallback) {
 
           mApi().forum.areas.read(thread.forumAreaId).callback(function(err, area) {
             thread.areaName = area.name;
-          });
+
 
           mApi().user.users.basicinfo.read(thread.creator).callback(function(err, user) {
             thread.creatorFullName = user.firstName + ' ' + user.lastName;
+            var d = new Date(thread.created);
+            thread.prettyDate = d.toLocaleString();
```

Comment:
js should use formatDate and dust template has it's own mechanism for formatting dates